source/components/hardware/hwxfsleep.c

```diff
@@ -180,13 +180,29 @@ AcpiHwSleepDispatch (
 #define ACPI_WAKE_PREP_FUNCTION_ID      1
 #define ACPI_WAKE_FUNCTION_ID           2
 
+/*
+ * Some compilers can handle designated initializers, which is needed
+ * under Linux kernel builds for structures that are entirely function
+ * pointers.
+ */
+#ifdef __GNUC__
+#define ACPI_SLEEP_FUNCTION_INIT(legacy, extended) \
+              { .legacy_function   = legacy,     \
+                .extended_function = extended }
```

Comment:
Anyway, I'm not sure what the benefit of doing this.
Even it is possible, we still cannot change the function definition order inside of ACPICA.
As it can break other compilers.